### PR TITLE
feat: add `pixi global edit`

### PR DESCRIPTION
**Pull request by ruben-arts**
Adds a subcommand to pixi global:
```
pixi global edit
```

This will open the manifest file in the editor defined in `EDITOR` path variable or defaults to `nano` or `notepad` depending on the machine. 

This code is stolen from `pixi config edit`

### CODE REVIEW

File: src/cli/global/edit.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    #[clap(flatten)]
+    config: ConfigCli,
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    let config = Config::with_cli_config(&args.config);
+    let project = Project::discover_or_create(args.assume_yes)
+        .await?
+        .with_cli_config(config.clone());
```
Comment (Hofer-Julian) [inline src/cli/global/edit.rs:24]:
```suggestion
    let project = Project::discover_or_create(args.assume_yes)
        .await?
        .with_cli_config(config.clone());
```

Just tried adding something invalid with `pixi global edit` and then running `pixi global edit` again. This line makes this fail.

File: src/task/file_hashes.rs [hunk 2]
```diff
@@ -111,7 +111,10 @@ impl FileHashes {
             .git_ignore(false)
             .git_global(false)
             .git_exclude(false)
-            .follow_links(true)
+            // Turn this back off as it can cause issues with symlinks:
+            // https://github.com/prefix-dev/pixi/issues/2196
+            // TODO: The current idea is to completely reimplement this without the `ignore` crate.
+            // .follow_links(true)
```
Comment (Hofer-Julian) [inline src/task/file_hashes.rs:117]: This change shouldn't show up here I think :)

Comment (ruben-arts) [reply src/task/file_hashes.rs:117]: There is a weird merge going on here.

File: src/cli/global/edit.rs (hunk 1 above)
Comment (ruben-arts) [reply src/cli/global/edit.rs:24]: hmm good point. Lets assume the file already exists. And otherwise just create it.

Comment (ruben-arts) [reply src/cli/global/edit.rs]: Changed that.

File: src/task/file_hashes.rs (hunk 2 above)
Comment (ruben-arts) [reply src/task/file_hashes.rs]: Non existent anymore

File: src/cli/global/edit.rs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    #[arg(env = "EDITOR")]
+    pub editor: Option<String>,
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    let manifest_path = Project::default_manifest_path()?;
+
+    let editor = args
+        .editor
+        .or_else(|| std::env::var("EDITOR").ok())
```
Comment (ruben-arts) [inline src/cli/global/edit.rs:21]:
```suggestion
```

File: src/cli/global/edit.rs [hunk 4]
```diff
@@ -1,41 +1,34 @@
...
-            "nano".to_string()
-        }
-        #[cfg(target_os = "windows")]
-        {
-            "notepad".to_string()
-        }
-    });
+    let editor = args
+        .editor
+        .or_else(|| std::env::var("EDITOR").ok())
```
Comment (Hofer-Julian) [inline src/cli/global/edit.rs]:
```suggestion
```
You already check for that with `clap`

**Review by Hofer-Julian — APPROVED** [review]
Works like a charm

**Pull request merged**